qiskit/ignis/verification/tomography/basis/gatesetbasis.py

```diff
@@ -0,0 +1,249 @@
+                            ) -> QuantumCircuit:
+        """
+        Creates a preperation circuit for the SPAM op
+
+        Params:
+            op: SPAM circuit name
+            qubit: qubit to be operated on
+
+        Returns:
+            The preperation circuit
+        """
+        circ = QuantumCircuit(qubit.register)
+        self.add_to_circuit(circ, qubit, op)
+        return circ
+
+    def preparation_matrix(self, label: str) -> np.array:
+        """
+        Returns the matrix corresponding to a gate label
+
+        Params:
+            label: Gate label
+
+        Returns:
+            The corresponding matrix
+        """
+        return self.gate_matrices[label]
+
+    def spam_matrix(self, label: str) -> np.array:
+        """
+        Returns the matrix corresponding to a spam label
+        Every spam is a sequence of gates, and so the result matrix
+        is the product of the matrices corresponding to those gates
+
+        Params:
+            label: Spam label
+
+        Returns:
+            The corresponding matrix
+        """
+        spec = self.spam_spec[label]
+        F_matrices = [self.gate_matrices[gate_label] for gate_label in spec]
+        result = functools.reduce(lambda a, b: a @ b, F_matrices)
+        return result
+
+    def get_tomography_basis(self) -> TomographyBasis:
+        """
+        Returns a TomographyBasis object
+        corresponding to the gate set tomography data
+
+        A TomographyBasis object should have
+        for both measurements and preperations
+        the sets of labels (the SPAM labels in our case),
+        circuit creation functions and corresponding matrices
+
+        Returns:
+            The gateset tomography data formatted as a TomographyBasis object
+        """
+        return TomographyBasis(self.name,
+                               measurement=(self.spam_labels,
+                                            self.measurement_circuit,
+                                            self.measurement_matrix),
+                               preparation=(self.spam_labels,
+                                            self.preparation_circuit,
+                                            self.preparation_matrix))
+
+
+def standard_gates_func(circ: QuantumCircuit,
+                        qubit: QuantumRegister,
+                        op: str
+                        ):
+    """
+        The gate creation function for the default set of gates
+        we use for gate set tomography:
+        Id and rotations by 90 degrees around the X and Y axis
+
+        Params:
+            circ: the circuit to add the gate to
+            qubit: qubit to be operated on
+            op: the gate name
+    """
+    if op == 'Id':
+        pass
+    if op == 'X_Rot_90':
+        circ.u2(-np.pi / 2, np.pi / 2, qubit)
+    if op == 'Y_Rot_90':
+        circ.u2(np.pi, np.pi, qubit)
+
+
+# PTM representation of Id
+G0 = np.array([[1, 0, 0, 0],
```

Comment:
It's not clear in rest of the documentation that these matrices should be PTM instead of unitary matrices.

Can these be passed in here `BaseOperator` objects (ie Operator, or one of the QuantumChannel reps), and then converted to `PTM` channel objects later when they are needed?

Reply:
This was meant to be temporary; I've now redone in 2617be2 the way `gateset_basis` is constructed and no matrix needs to be explicitly passed anymore.